blob/gcsblob/gcsblob.go

```diff
@@ -37,13 +49,57 @@ import (
 	"github.com/google/go-cloud/gcp"
 
 	"cloud.google.com/go/storage"
+	"golang.org/x/oauth2/google"
 	"google.golang.org/api/googleapi"
 	"google.golang.org/api/iterator"
 	"google.golang.org/api/option"
 )
 
 const defaultPageSize = 1000
 
+func init() {
+	blob.Register("gs", func(ctx context.Context, u *url.URL) (driver.Bucket, error) {
+		q := u.Query()
+		opts := &Options{}
+
+		if accessID := q["access_id"]; len(accessID) > 0 {
+			opts.GoogleAccessID = accessID[0]
+		}
+
+		if keyPath := q["private_key_path"]; len(keyPath) > 0 {
+			pk, err := ioutil.ReadFile(keyPath[0])
+			if err != nil {
+				return nil, fmt.Errorf("failed to read private key: %v", err)
```

Comment:
nit: the usual Go style is `"reading private key: %v", err`.  In other words, we don't usually say "failed" or "error while...". 

If you've adopted a different convention for Go Cloud, that's fine; it's not a big deal.

Reply:
I should also mention that if something like the Go 2 error-handling proposal is adopted, it will be much easier to have one message for the whole function, so maybe we shouldn't try to be too specific here. Of course, we shouldn't consider the error message to be part of our API, so we can always change it later.